src/executor/Executor.cpp

```diff
@@ -113,40 +113,23 @@ Executor *Executor::makeExecutor(const PlanNode *node,
     }
 
     Executor *exec = makeExecutor(qctx, node);
-    switch (node->dependencies().size()) {
-        case 0: {
-            // Do nothing
-            break;
-        }
-        case 1: {
-            if (node->kind() == PlanNode::Kind::kSelect) {
-                auto select = asNode<Select>(node);
-                auto thenBody = makeExecutor(select->then(), qctx, visited);
-                auto elseBody = makeExecutor(select->otherwise(), qctx, visited);
-                auto selectExecutor = static_cast<SelectExecutor *>(exec);
-                selectExecutor->setThenBody(thenBody);
-                selectExecutor->setElseBody(elseBody);
-            } else if (node->kind() == PlanNode::Kind::kLoop) {
-                auto loop = asNode<Loop>(node);
-                auto body = makeExecutor(loop->body(), qctx, visited);
-                auto loopExecutor = static_cast<LoopExecutor *>(exec);
-                loopExecutor->setLoopBody(body);
-            }
-            auto dep = makeExecutor(node->dep(0), qctx, visited);
-            exec->dependsOn(dep);
-            break;
-        }
-        case 2: {
-            auto left = makeExecutor(node->dep(0), qctx, visited);
-            auto right = makeExecutor(node->dep(1), qctx, visited);
-            exec->dependsOn(left)->dependsOn(right);
-            break;
-        }
-        default: {
-            LOG(FATAL) << "Unsupported plan node type which has dependencies: "
-                       << node->dependencies().size();
-            break;
-        }
+
+    if (node->kind() == PlanNode::Kind::kSelect) {
+        auto select = asNode<Select>(node);
+        auto thenBody = makeExecutor(select->then(), qctx, visited);
+        auto elseBody = makeExecutor(select->otherwise(), qctx, visited);
+        auto selectExecutor = static_cast<SelectExecutor *>(exec);
+        selectExecutor->setThenBody(thenBody);
+        selectExecutor->setElseBody(elseBody);
+    } else if (node->kind() == PlanNode::Kind::kLoop) {
+        auto loop = asNode<Loop>(node);
+        auto body = makeExecutor(loop->body(), qctx, visited);
+        auto loopExecutor = static_cast<LoopExecutor *>(exec);
+        loopExecutor->setLoopBody(body);
+    }
+
+    for (size_t i = 0; i < node->numDeps(); ++i) {
```

Comment:
I think the iterator over container is better than the index.

Reply:
Plan node already have `dep()` function, needs not to provide a `dependencies()` to use.